state/protocol/badger/state.go

```diff
@@ -46,6 +46,10 @@ type State struct {
 	// larger than the height of the root block of the spork, (also cached below as
 	// `sporkRootBlockHeight`), for instance if the node joined in an epoch after the last spork.
 	rootHeight uint64
+	// sealedRootHeight returns the root block that is sealed.
+	// For non-execution node, sealedRootHeight == rootHeight == rootSnapshot.SealingSegment.Highest().Header.Height
+	// For execution node, sealedRootHeight == rootSnapshot.SealingSegment.Sealed().Header.Height < rootHeight == rootSnapshot.SealingSegment.Highest().Header.Height
```

Comment:
I don't understand why we would use different and inconsistent representations for this value, depending on the role. From my perspective, the reason for adding the extra field `sealedRootHeight` is to avoid the need for different nodes using inconsistent representations for the root block.